tests/unit/test_all_paths.cpp

```diff
@@ -307,3 +307,67 @@ TEST_F(AllPathsTest, DestinationSpecificPaths) {
 	Graph_Free(g);
 }
 
+
+// Test all paths of the minimum length from source to a specific destination node.
+TEST_F(AllPathsTest, OnlyShortestPaths) {
+	NodeID p00_0[2] = {1, 0};
+
+	NodeID *p[1] = {p00_0};
+
+	Graph *g = BuildGraph();
+
+	NodeID srcNodeID = 0;
+	Node src;
+	Path *path = NULL;
+	Graph_GetNode(g, srcNodeID, &src);
+	unsigned int minLen = 0;
+	unsigned int maxLen = UINT_MAX - 2;
+	unsigned int pathsCount = 0;
+	int relationships[] = {GRAPH_NO_RELATION};
+	AllPathsCtx *ctx = AllPathsCtx_New(&src, &src, g, relationships, 1, GRAPH_EDGE_DIR_OUTGOING,
+									   minLen, maxLen, NULL, NULL, 0, true);
+
+	while((path = AllPathsCtx_NextPath(ctx))) {
+		ASSERT_TRUE(pathArrayContainsPath(p, 1, path));
+		pathsCount++;
+	}
+
+	ASSERT_EQ(pathsCount, 1);
+
+	AllPathsCtx_Free(ctx);
+	Graph_Free(g);
+}
+
+// Test all paths of the minimum length from source to a specific destination
+// node with a minimum length
+TEST_F(AllPathsTest, OnlyShortestPathsMinLength) {
```

Comment:
can't these two test be covered by a flow test ?

Reply:
They definitely can be, but since there was a unit test covering this API I thought it would be sensible for it to cover the extension to the API. Would you prefer this to be deleted and rewritten as a flow test?